Warn on the use of console

ashur opened the pull request:
## Summary

- Adds `no-console` rule as `warn`, warning on use of `console` in all cases
- Adds a `CHANGELOG.md` and a pull request template to the project

Adding `no-console` as a warning could help catch any debugging uses of `console` calls before being committed or pushed.

## Usage

In cases where a console call is desired in non-debugging contexts, the rule can be disabled:

```javascript
// eslint-disable-next-line no-console
console.log("This message should be logged to an end-user's browser");
```

## Tasks

- [x] Update CHANGELOG

## Notes

ESLint on `console` use:

> In JavaScript that is designed to be executed in the browser, it’s considered a best practice to avoid using methods on console. Such messages are considered to be for debugging purposes and therefore not suitable to ship to the client.

## Links

- https://eslint.org/docs/latest/rules/no-console

ashur committed:
Add CHANGELOG

ashur committed:
Add "no-console"

https://eslint.org/docs/latest/rules/no-console

jakefentress commented:
@ashur I like the no-console for commits, but this might keep things from compiling locally so you couldn't use a console.log for debugging, either. But, that may be more a function of how we're currently using eslint in the bootstrap as opposed to maybe how it should be used.

ashur committed:
Update no-console to "warn"

We don't want to break existing uses of eslint-config-fk in other projects, and setting as an error could prevent development builds in some cases

ashur commented:
@jacobfentress Great point! I dropped the rule to `warn` instead of `error` to prevent breaking existing uses of `eslint-config-fk`.

jakefentress reviewed (approved):
This is a good addition.

One request, though I'm approving this now, would be to add the pull request template to remind folks to update the changelog since you are introducing it here.

ashur committed:
Add initial PR template
